R/parse_config.R

```diff
@@ -0,0 +1,355 @@
+          arrange(time) # arrange at the end ensures that extracted messages are stored in the order they appear in continuous time rather than the order they were requested (i.e. the j iterator)
+      }
+
+      extracted_msgs <- extracted_msgs$et.msg
+
+
+      # if(length(extracted_msgs) != length(check_these)){
+      # N.B. initially, this was just to set up the df in instances in which the expected msgs and extracted msgs did not match, and to flag missing messages with an NA for later inspection.
+      # now, I think this is a more general way to setup such a comparison df.
+
+      # output df will be ordered according to the ordering of expected messages encoded in check_these. Though, extracted will be mutated "to" the user's expectations, thus the ordering of extracted messages could be incorrect.
+      cvec <- sapply(check_these, function(x) grepl(x, extracted_msgs))
+      df <- data.frame(apply(cvec, 2, function(x) any(x))) %>% rownames_to_column() # right column will throw FALSE if there are no matches in the extracted strings, meaning the message is missing.
+      colnames(df) <- c("requested", "match")
+      df <- df %>% group_by(requested) %>%  mutate(extracted = ifelse(match, extracted_msgs[grepl(requested, extracted_msgs)], NA)) %>% ungroup()
+
+      # }
+
+      # now, if requested, check the ordering and get mad at user if extracted msgs are out of the expected order.
+      # do this by utilizing the initial vector of extracted messages.
+      if(c.e$event_info$msg_seq$ordered){
+        # this is almost certainly error-prone, can come back to this later. on first glance it seems like it could work in the mean time.
+        df <- df %>% left_join(data.frame(extracted = extracted_msgs,
+                                          order_check = extracted_msgs == na.omit(df$extracted)), by = "extracted")
+
+      } else {
+        df <- df %>% mutate(order_check = TRUE) # don't cause any trouble.
+      }
+
+      df <- df %>% mutate(eventn = unique(eblock$eventn),
+                          block = unique(eblock$block),
+                          block_trial = unique(eblock$block_trial),
+                          event = unique(eblock$event)) %>% select(eventn, requested, extracted, match,order_check, block, block_trial, event)
+
+      # print(df)
+      if(!all(df$match) | !all(df$order_check)){
+        seq_errs[[i]] <- df
+      }
+    }
+
+    # store imperfect seq matches and print warning. Update to include empty list if all successful and just print warning if there is an issue.
+    eye$metadata[["msg_seq_errs"]] <- seq_errs
+
+    if(length(seq_errs) != 0){
+      warning("IMPERFECT MESSAGE SEQUENCES FOUND IN: 'ep.eye$metadata$msg_seq_errs'")
+    }
+
+
+  }, describe_text = dt)
+
+  return(eye)
+}
+
+
+#' split off function for extracting event info
+#'
+get_event_info <- function(c.e, eye, dt){
+  cat(dt)
+  tryCatch.ep({
```

Comment:
Maybe more a matter of style, but I usually try to avoid long expressions inside a tryCatch since it makes the code harder to read (like, where does the tryCatch end?), including indentation. I find it easier to use a small worker subfunction and to tryCatch a call to that function.